async_producer.go

```diff
@@ -555,6 +555,21 @@ func (tp *topicProducer) partitionMessage(msg *ProducerMessage) error {
 		return err
 	}
 
+	// Select only partitions with leaders in this rack if configured so, falling back if none are available.
+	if tp.parent.conf.Producer.PartitionerRackAware {
+		var availablePartitionsInRack []int32
```

Comment:
🤔 The “available” prefix here I think might be over disambiguation? That is `partitionsInRack` is just as equally clear as to purpose.